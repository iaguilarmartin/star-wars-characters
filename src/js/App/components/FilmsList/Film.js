import React from 'react';
import { format } from 'date-fns';

import filmShape from './film-shape';
import CalendarIcon from './icons/Calendar';
import MoneyIcon from './icons/Money';
import FilmIcon from './icons/Film';
import {
  Article,
  Container,
  Description,
  ReleaseDate,
  Info,
  Title
} from './Film.styles';

const Film = ({
  film: {
    title,
    director,
    producer,
    release_date: releaseDate,
    opening_crawl: openingCrawl
  }
}) => (
  <Article>
    <Container>
      <Title>{title}</Title>
      <ReleaseDate>
        <CalendarIcon />
        {format(new Date(releaseDate), 'dddd, MMM D YYYY')}
      </ReleaseDate>
      <Info>
        <FilmIcon />
        {director}
      </Info>
      <Info>
        <MoneyIcon />
        {producer}
      </Info>
      <Description>{openingCrawl}</Description>
    </Container>
  </Article>
);

Film.propTypes = {
  film: filmShape.isRequired
};

export default Film;
