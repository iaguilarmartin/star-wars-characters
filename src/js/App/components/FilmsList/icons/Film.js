import React from 'react';

import colors from '../../../styles/colors';

const FilmIcon = () => (
  <svg version="1.1" viewBox="0 0 22 26" xmlns="http://www.w3.org/2000/svg">
    <g fill="none" fillRule="evenodd">
      <g transform="translate(-427 -46)" fill={colors.white} fillRule="nonzero">
        <g transform="translate(40 46)">
          <g transform="translate(387)">
            <path d="m21 0h-20c-0.55 0-1 0.45-1 1v24c0 0.55 0.45 1 1 1h20c0.55 0 1-0.45 1-1v-24c0-0.55-0.45-1-1-1zm-3 20v-2h3v2h-3zm3 2v2h-3v-2h3zm-3-6v-2h3v2h-3zm0-4v-2h3v2h-3zm0-4v-2h3v2h-3zm0-4v-2h3v2h-3zm-13 8v-10h12v10h-12zm12 2v10h-12v-10h12zm-16 6v-2h3v2h-3zm3 2v2h-3v-2h3zm-3-6v-2h3v2h-3zm0-4v-2h3v2h-3zm0-4v-2h3v2h-3zm0-4v-2h3v2h-3z" />
            <path d="m17 2h-12v10h12v-10zm0 12h-12v10h12v-10z" opacity=".3" />
          </g>
        </g>
      </g>
    </g>
  </svg>
);

export default FilmIcon;
