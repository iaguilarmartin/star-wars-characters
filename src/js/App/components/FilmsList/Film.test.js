import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import mockFilm from '../../../../../jest/mocks/film';

import Film from './Film';

describe('<Film/>', () => {
  it('renders', () => {
    shallow(<Film film={mockFilm} />);
  });

  it('matches the snapshot', () => {
    const wrapper = shallow(<Film film={mockFilm} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('displays date in "dddd, MMM D YYYY" format', () => {
    const film = {
      ...mockFilm,
      release_date: '1980-05-18'
    };
    const wrapper = shallow(<Film film={film} />);
    expect(wrapper.find('ReleaseDate').text()).toContain(
      'Saturday, May 17 1980'
    );
  });
});
