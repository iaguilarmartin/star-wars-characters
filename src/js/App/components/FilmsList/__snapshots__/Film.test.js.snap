// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Film/> matches the snapshot 1`] = `
<Article>
  <Container>
    <Title>
      The Empire Strikes Back
    </Title>
    <ReleaseDate>
      <CalendarIcon />
      Friday, May 16 1980
    </ReleaseDate>
    <Info>
      <FilmIcon />
      Irvin Kershner
    </Info>
    <Info>
      <MoneyIcon />
      Gary Kurtz, Rick McCallum
    </Info>
    <Description>
      It is a dark time for the
Rebellion. Although the Death
Star has been destroyed,
Imperial troops have driven the
Rebel forces from their hidden
base and pursued them across
the galaxy.

Evading the dreaded Imperial
Starfleet, a group of freedom
fighters led by Luke Skywalker
has established a new secret
base on the remote ice world
of Hoth.

The evil lord Darth Vader,
obsessed with finding young
Skywalker, has dispatched
thousands of remote probes into
the far reaches of space....
    </Description>
  </Container>
</Article>
`;
